File: Client/src/app/models/product.ts
import { Article } from './article';

/**
 * Represents a product in the system.
 */
export interface Product {
  /**
   * Unique identifier for the product.
   */
  id: number;

  /**
   * The brand name of the product.
   */
  brandName: string;

  /**
   * The specific name of the product.
   */
  name: string;

  /**
   * A list of articles related to this product.
   */
  article: Article[];
}

File: API/Model/Product.cs
namespace Flaschenpost_SE.Model;

/// <summary>
///     Represents a product in the system.
/// </summary>
public class Product
{
    /// <summary>
    ///     Gets or sets the product's unique identifier.
    /// </summary>
    public int Id { get; set; }

    /// <summary>
    ///     Gets or sets the brand name of the product.
    /// </summary>
    public string BrandName { get; set; }

    /// <summary>
    ///     Gets or sets the name of the product.
    /// </summary>
    public string Name { get; set; }

    /// <summary>
    ///     Gets or sets the list of articles associated with this product.
    /// </summary>
    public List<Article> Articles { get; set; }
}